### ISSUE TITLE
Attention option in recognition module

### ISSUE THREAD

**Issue by Serjio42** [JaidedAI/EasyOCR#1101]
Hi.
I see 2 options (CTC and Attention) in the model [constructor](https://github.com/JaidedAI/EasyOCR/blob/f947eaa36a55adb306feac58966378e01cc67f85/trainer/model.py#L44):
```
if opt.Prediction == 'CTC':
            self.Prediction = nn.Linear(self.SequenceModeling_output, opt.num_class)
        elif opt.Prediction == 'Attn':
            self.Prediction = Attention(self.SequenceModeling_output, opt.hidden_size, opt.num_class)
```
As I know, there are 2 typical options that are commonly used in recognition models: RNN and Self-attention.
But if you look in class [AttentionCell](https://github.com/JaidedAI/EasyOCR/blob/master/trainer/modules/prediction.py#L61) you can see nn.LSTMCell inside it. Looks a bit confusing. In addition, what about using CTC loss in Self-attention mode?.. As you can see in abstract in [this](https://arxiv.org/abs/2104.07787.pdf) paper "a Self-Attention encoder coupled with the CTC decoder... outperforms all the others".

For clarity, can you please clarify the type of attention (and maybe paper or documentation on it) used in EasyOCR implementation? Maybe we can extend it to reproduce the design mentioned in the paper (from the paragraph above)?
Thanks.
